src/libcore/marker.rs

```diff
@@ -311,16 +311,17 @@ impl<T: ?Sized> Clone for ContravariantType<T> {
 ///
 /// # Example
 ///
-/// The Cell type is an example which uses unsafe code to achieve
-/// "interior" mutability:
+/// The Cell type is an example of an `InvariantType` which uses unsafe
+/// code to achieve "interior" mutability:
 ///
 /// ```
 /// struct Cell<T> { value: T }
 /// ```
 ///
-/// The type system would infer that `value` is only read here and
-/// never written, but in fact `Cell` uses unsafe code to achieve
-/// interior mutability.
+/// The type system would infer that `value` is only read here
+/// and never written, but in fact `Cell` uses unsafe code to achieve
+/// interior mutability. In order to get correct behavior, the
+/// `InvariantType` marker must be applied.
```

Comment:
This is not strictly true. That is, *all* types must use `unsafe` code to achieve interior mutability. And if they do so, types should use an `InvariantType` marker (though this is likely to be replaced; see RFC https://github.com/rust-lang/rfcs/pull/738). The only exception to that last rule is `Cell`, which has no marker because it is builtin to the compiler.